src/VisualStudio/Core/Def/Implementation/FindUsages/ItemOriginColumnDefinition.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Composition;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+using Microsoft.VisualStudio.Utilities;
+
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.FindUsages
+{
+    /// <summary>
+    /// Custom column to display the item's origin for the Find All References window.
+    /// </summary>
+    [Export(typeof(ITableColumnDefinition))]
+    [Name(ColumnName)]
+    internal class ItemOriginColumnDefinition : TableColumnDefinitionBase
+    {
+        public const string ColumnName = StandardTableKeyNames.ItemOrigin;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public ItemOriginColumnDefinition()
+        {
+        }
+
+        public override bool IsFilterable => true;
+        public override string Name => ColumnName;
+        public override string DisplayName => ServicesVSResources.Item_origin; // TODO: Localize
+
+        public override bool TryCreateStringContent(ITableEntryHandle entry, bool truncatedText, bool singleColumnView, out string content)
+        {
+            if (entry.TryGetValue(Name, out ItemOrigin origin))
```

Comment:
do we expect the origin column to be visible to users?  or wuld we only want a 'repository' column shown to them?

Reply:
Can we have it available, but hidden by default?
For build, we probably don't need it. Once we fully support local \ current repo \ current organization \ everything, it would be good to make this available to the users. Feedback will tell whether users need it or not

Reply:
i'm happy to try that.  i'm just not sure how to actually do that.  i can't find any api to change the visbility.  do you do that on your end (Since i presume you own the definition of that column).

Reply:
oh, n.m.  it's on our end.  done.